src/config.rs

```diff
@@ -282,11 +282,12 @@ impl<'a> Cfg<'a> {
         let default_host_triple =
             settings_file.with(|s| Ok(get_default_host_triple(s, process)))?;
         // Environment override
-        let env_override = non_empty_env_var("RUSTUP_TOOLCHAIN", process)?
-            .map(ResolvableLocalToolchainName::try_from)
-            .transpose()?
-            .map(|t| t.resolve(&default_host_triple))
-            .transpose()?;
+        let env_override = match process.var_opt("RUSTUP_TOOLCHAIN")? {
+            Some(tc) => {
+                Some(ResolvableLocalToolchainName::try_from(&tc)?.resolve(&default_host_triple)?)
+            }
+            None => None,
```

Comment:
The `Some(_) => Some(_)` and `None => None` branches are a little ugly, but this saves on the dual `.map().transpose()?` calls so I'm inclined to still prefer this.